marketplace-service/src/main/java/com/axonivy/market/service/impl/VersionServiceImpl.java

```diff
@@ -96,6 +103,19 @@ public List<MavenArtifactVersionModel> getArtifactsAndVersionToDisplay(String pr
     return results;
   }
 
+  @Override
+  public ProductJsonContent getProductJsonContentFromNameAndTag(String name, String tag) {
+    return productJsonContentRepository.findByNameAndTag(name, tag);
+  }
+
+  @Override
+  public List<String> getVersionsForDesigner(String productId, Boolean isShowDevVersion, String designerVersion) {
+    Product product = productRepository.findById(productId).orElse(null);
```

Comment:
If you implement it like that, we may miss the new version which has just been released after the sync function is finished.
I case all of version have been processed, you can reuse one of 2 props: [version is [maven ](https://github.com/axonivy-market/marketplace/blob/master/marketplace-service/src/main/java/com/axonivy/market/entity/MavenArtifactVersion.java) or ](https://github.com/axonivy-market/marketplace/blob/1ed484a6808800e53ce9793eb39b34b514d2ffc2/marketplace-service/src/main/java/com/axonivy/market/entity/MavenArtifactVersion.java#L30) or https://github.com/axonivy-market/marketplace/blob/1ed484a6808800e53ce9793eb39b34b514d2ffc2/marketplace-service/src/main/java/com/axonivy/market/entity/Product.java#L63C11-L63C15

Reply:
Please use productRepository.getReleasedVersionsById() to get the list of versions. This prop is already updated in the sync feature and prevents us from getting all of unnecessary content module content when we just need to get it tags